xmake/core/project/target.lua

```diff
@@ -953,6 +953,7 @@ function _instance:orderpkgs(opt)
                 end
             end
         end
+        table.sort(packages, function (a, b) return a:name() < b:name() end)
```

Comment:
we should not sort it, it will break package links order

Reply:
from my understanding, this table comes from `project.required_packages()` which calls `project._load_requires()` whichs doesn't seem to sort packages?

Those are xmake internals so I'm not sure at all, the only things I'm sure is that this table is not stable.

Reply:
Even if it is unstable, it is at least guaranteed to be in the correct linking order. Sorting it would completely break the linking order of the packages.

Reply:
maybe package deps should be sorted by name then?

Reply:
no, we cannot sort it.

Reply:
In fact I'm not sure where the order is lost, since package deps are already given in a specific order.

Reply:
I removed it, it wasn't necessary